pkg/sdk/testint/streams_gen_integration_test.go

```diff
@@ -18,37 +19,93 @@ func TestInt_Streams(t *testing.T) {
 	databaseId := testClientHelper().Ids.DatabaseId()
 	schemaId := testClientHelper().Ids.SchemaId()
 
-	assertStream := func(t *testing.T, s *sdk.Stream, id sdk.SchemaObjectIdentifier, sourceType string, mode string) {
-		t.Helper()
-		assert.NotNil(t, s)
-		assert.Nil(t, s.TableOn)
-		assert.Equal(t, id.Name(), s.Name)
-		assert.Equal(t, databaseId.Name(), s.DatabaseName)
-		assert.Equal(t, schemaId.Name(), s.SchemaName)
-		assert.Equal(t, "some comment", *s.Comment)
-		assert.Equal(t, sourceType, *s.SourceType)
-		assert.Equal(t, mode, *s.Mode)
-	}
-
+	// There is no way to check at/before fields in show and describe. That's why in Create tests we try creating with these values, but do not assert them.
 	t.Run("CreateOnTable", func(t *testing.T) {
-		table, cleanupTable := testClientHelper().Table.CreateTableInSchema(t, schemaId)
+		table, cleanupTable := testClientHelper().Table.CreateWithChangeTracking(t)
 		t.Cleanup(cleanupTable)
+		tableId := table.ID()
+
+		tag, tagCleanup := testClientHelper().Tag.CreateTag(t)
+		t.Cleanup(tagCleanup)
 
 		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
-		req := sdk.NewCreateStreamOnTableRequest(id, table.ID()).WithComment(sdk.String("some comment"))
+		req := sdk.NewCreateOnTableStreamRequest(id, tableId).
+			WithOn(*sdk.NewOnStreamRequest().WithAt(true).WithStatement(*sdk.NewOnStreamStatementRequest().WithOffset("0"))).
+			WithAppendOnly(true).
+			WithShowInitialRows(true).
+			WithComment("some comment").
+			WithTag([]sdk.TagAssociation{
+				{
+					Name:  tag.ID(),
+					Value: "v1",
+				},
+			})
 		err := client.Streams.CreateOnTable(ctx, req)
 		require.NoError(t, err)
-		t.Cleanup(func() {
-			err := client.Streams.Drop(ctx, sdk.NewDropStreamRequest(id))
-			require.NoError(t, err)
-		})
+		t.Cleanup(testClientHelper().Stream.DropFunc(t, id))
 
-		s, err := client.Streams.ShowByID(ctx, id)
+		tag1Value, err := client.SystemFunctions.GetTag(ctx, tag.ID(), id, sdk.ObjectTypeStream)
 		require.NoError(t, err)
+		assert.Equal(t, "v1", tag1Value)
 
-		// TODO [SNOW-1348112]: make nicer during the stream rework
-		assert.Equal(t, table.ID().FullyQualifiedName(), sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(*s.TableName).FullyQualifiedName())
-		assertStream(t, s, id, "Table", "DEFAULT")
+		assertions.AssertThatObject(t, objectassert.Stream(t, id).
+			HasName(id.Name()).
+			HasDatabaseName(id.DatabaseName()).
+			HasSchemaName(id.SchemaName()).
+			HasComment("some comment").
+			HasSourceType("Table").
+			HasMode("APPEND_ONLY").
```

Comment:
are there any enums for sourceType and mode? Asserts could be extended to use them

Reply:
These will be added in the next PR.